dandiapi/analytics/tasks/__init__.py

```diff
@@ -0,0 +1,85 @@
+from collections import Counter
+from pathlib import Path
+from typing import Generator
+
+from celery.app import shared_task
+from django.conf import settings
+from django.db import transaction
+from django.db.models.aggregates import Max
+from django.db.models.expressions import F
+from s3logparse import s3logparse
+
+from dandiapi.analytics.models import ProcessedS3Log
+from dandiapi.api.models.asset import AssetBlob, EmbargoedAssetBlob
+from dandiapi.api.storage import get_boto_client, get_embargo_storage, get_storage
+
+# should be one of the DANDI_DANDISETS_*_LOG_BUCKET_NAME settings
+LogBucket = str
+
+
+def _bucket_objects_after(bucket: str, after: str | None) -> Generator[dict, None, None]:
+    assert bucket in [
+        settings.DANDI_DANDISETS_LOG_BUCKET_NAME,
+        settings.DANDI_DANDISETS_EMBARGO_LOG_BUCKET_NAME,
+    ]
+    embargoed = bucket == settings.DANDI_DANDISETS_EMBARGO_LOG_BUCKET_NAME
+    s3 = get_boto_client(get_storage() if not embargoed else get_embargo_storage())
+    kwargs = {}
+    if after:
+        kwargs['StartAfter'] = after
```

Comment:
that is great -- I forgot or didn't know that it is possible to do that since we have millions of keys/files there already.  

With that in mind -- are you planing to run it first fetching all those millions of logs first?   as I have mentioned on slack - it would be millions and TBs of log files. Would such job have enough time?

Reply:
If we do an initial migration I was planning on doing that out of band, this was just for the incremental job. I haven't yet looked at the price implications of the initial migration, if you're already downloading the historical logs for backup purposes it might make sense to have you run a one liner against them instead.